src/google-maps/map-polyline/map-polyline.ts

```diff
@@ -0,0 +1,234 @@
+   */
+  @Output() mapDragend = new EventEmitter<google.maps.MouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.dragstart
+   */
+  @Output() mapDragstart = new EventEmitter<google.maps.MouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.mousedown
+   */
+  @Output() mapMousedown = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.mousemove
+   */
+  @Output() mapMousemove = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.mouseout
+   */
+  @Output() mapMouseout = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.mouseover
+   */
+  @Output() mapMouseover = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.mouseup
+   */
+  @Output() mapMouseup = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.rightclick
+   */
+  @Output() mapRightclick = new EventEmitter<google.maps.PolyMouseEvent>();
+
+  private readonly _options = new BehaviorSubject<google.maps.PolylineOptions>({});
+  private readonly _path =
+      new BehaviorSubject<google.maps.MVCArray<google.maps.LatLng>|google.maps.LatLng[]|
+                          google.maps.LatLngLiteral[]|undefined>(undefined);
+
+  private readonly _destroy = new Subject<void>();
+
+  private readonly _listeners: google.maps.MapsEventListener[] = [];
+
+  _polyline?: google.maps.Polyline;
+
+  constructor(private readonly googleMap: GoogleMap) {}
+
+  ngOnInit() {
+    const combinedOptionsChanges = this._combineOptions();
+
+    combinedOptionsChanges.pipe(take(1)).subscribe(options => {
+      this._polyline = new google.maps.Polyline(options);
+      this._polyline.setMap(this.googleMap._googleMap);
+      this._initializeEventHandlers();
+    });
+
+    this._watchForOptionsChanges();
+    this._watchForPathChanges();
+  }
+
+  ngOnDestroy() {
+    this._destroy.next();
+    this._destroy.complete();
+    for (let listener of this._listeners) {
+      listener.remove();
+    }
+    if (this._polyline) {
+      this._polyline.setMap(null);
+    }
+  }
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.getDraggable
+   */
+  getDraggable(): boolean {
+    return this._polyline!.getDraggable();
+  }
+
+  /**
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Polyline.getEditable
+   */
+  getEditable(): boolean {
+    return this._polyline!.getEditable();
```

Comment:
From what I can tell, we aren't guaranteed to have a `_polyline` at all times so the the `!` here is misleading. It might be better to null check it. Same goes for all the other getters.

Reply:
I did it like this because _polyline is initialized in ngOnInit, but it should always be initialized. Let me know if the latest changes I'm uploading are acceptable.